### DISCUSSION TITLE
Rock Paper Scissors: <Lessons don't guide you to the project scope>

### DISCUSSION THREAD

**Discussion by Chronicide** [TheOdinProject/curriculum#26344] [General]
### Describe your suggestion

The lessons included before the first JS project do not lead you to the project in a meaningful way
- even having some python experience, this is just confusing
- There need to be a more comprehensive lesson(s) about linking .js files and running general testing and setting up functions that are relevant to the project like the CSS section does instead of long videos on youtube that are linked

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-rock-paper-scissors

### Checks

- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### (Optional) Additional Comments

I love this project as a whole so far, thank you for the hard work that goes into this

**Comment by rlmoser99** [comment]
Can you please provide more clarification on what you struggled with? You linked the first JS project, but I am not sure what you mean by several things. Such as what long videos you are referring to and what you mean by running general testing.

Without this additional information, my answer is that our curriculum will often have topics that learners must research on their own. Our lessons do not provide every single piece of information that a person will need to know, which helps equip them with the skills to do their research when they want to learn how to do something new.

**Comment by Chronicide** [comment]
Thank you for the reply! I was really struggling whit functions, specifically using parameters and arguments:
function playGame(playerSelection, computerSelection){}
- I was having a hard time going back through the JS sections prior to find what I needed as i was told to use these 2 parameters, when you read about it in a small "a + b" example it makes sense but seeing it in the code was much harder to grasp when you have to write all from scratch without using it before.

The best way to express it i think is using the FlexBox section before the landing page assignment here: https://www.theodinproject.com/lessons/foundations-alignment
- I really love how you work through several small exercises to use what is provided as a framework, Although i still really struggled at points, it gave a great sequential order to start the landing page project.
- I would have those "Oh yeah I had to make the image above the text in the cards exercise, let me take a look at my code to see how i dealt with that" or the documentation pages before had great visual guides and most of the content needed to succeed was part of TheOdinProject pages themselves, I knew flexbox axes would be in that category. Hopefully that helps explain it a little better!
- I know the struggle is part of learning but I really had no idea even how to understand what was wrong rather then being close and needing a little refresher

**Comment by haplorhine** [comment]
Hello. I also struggled a lot with this seemingly simple project especially with adapting the provided boilerplate code to still follow the bullet points of the assignment. 

**Here I want to list a few things I didn't understand or which I have issues with:**

Point 2 of the instructions states "This game is going to be played completely from the console[...]", which is in contrast to point 6 which states "Use `prompt()` to get input from the user."
This prompt method opens a dialog.

Point 4 doesn't clearly state what the `playRound` function should do, only that it should return a String (i.e. "You Lose! Paper beats Rock"). Regarding the following clean code lesson it isn't clear that a function called "playRound" returns a String. Also from the text it isn't clear where the score incrementation should happen. It seems like it should happen in `playRound` but then you have a function that returns a String and has side effects.
The sub point of point 4 states "Make your function’s playerSelection parameter case-insensitive". So this String manipulations stuff is another thing that takes place in `playRound`. This function seems to be doing a lot.

Point 6 instructs to "[w]rite a NEW function called `game()`". The following clean code lesson says: "Variables should always begin with a noun or an adjective [...] and functions with a verb." The word "game" can be a verb but I suppose it's mostly used as a noun at least for non native english speakers.
It is also quite unclear how the non-loop version should be implemented. In the loop, one can use the provided declaration of `const computerSelection = getComputerChoice();`. In the non loop version you need differently named const variables like `const computerSelectionRound1`, `const computerSelectionRound2` ... which isn't clear from the description.

**Comment by rlmoser99** [comment]
Thanks for these clarifications and thoughts. Since this issue is not immediately actionable, I am going to convert this to a discussion. This will allow the @TheOdinProject/foundations team to review your ideas and figure out what changes (if any) need to be made.
